### PR TITLE
Adds support for using JS files as a mock

### PR DESCRIPTION
**Pull request by joeyciechanowicz**
You can now use a `.js` file as a mock, allowing you to share common response types between mocks.

Also fixes the JSDoc types to all be the same casing 🎈

### CODE REVIEW

File: __tests__/unit/mock-backend-response.test.js [hunk 1]
```diff
@@ -26,76 +35,62 @@ describe('Render Backend Response', () => {
...
-		const next = jest.fn();
-		middleware(request, undefined, next);
-
-		__getReadFileCallback()({error: 'not found'});
+			path: '/does/not/exist'
+		}, undefined, next);
 
 		expect(next).toHaveBeenCalledWith();
+		expect(next.mock.calls[0].length).toEqual(0);
 	});
```
Comment (morgaan) [inline __tests__/unit/mock-backend-response.test.js]: Isn't it two way of testing the same thing, i.e that `next` have been called with no arguments?

File: examples/mocking/mocking.js [hunk 2]
```diff
@@ -1,27 +1,19 @@
...
-if (process.env.NODE_ENV === 'development') {
-	console.log('Adding mocking');
-	app.use(mockBackendResponse({
-		directory: "./mocks"
-	}));
-}
-
-app.use(backendProxy({backend: 'http://localhost:8080/api'}));
-app.use(renderBackendResponse());
+app.use((request, response) => {
```
Comment (morgaan) [inline examples/mocking/mocking.js]: Here we remove the `if (process.env.NODE_ENV === 'development') {` check. What was the reasoning behind it?

File: src/mock-backend-response.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-const fs = require('fs');
+// const fs = require('fs');
```
Comment (morgaan) [inline src/mock-backend-response.js]: Shall we remove this obsolete line?

File: __tests__/unit/mock-backend-response.test.js [hunk 4]
```diff
@@ -26,76 +35,62 @@ describe('Render Backend Response', () => {
...
-			statusCode: 500,
-			message: expect.stringContaining('Error de-serialising mock response')
-		}));
+		expect(() => {
+			middleware({
+				method: 'get',
+				path: 'runtime-error'
+			}, undefined, next);
+		}).toThrow('Unhandled error');
 	});
```
Comment (morgaan) [inline __tests__/unit/mock-backend-response.test.js]: Is it worth to add a `expect(next).toHaveBeenCalledTimes(0)`?

File: __tests__/unit/mock-backend-response.test.js (hunk 1 above)
Comment (joeyciechanowicz) [reply __tests__/unit/mock-backend-response.test.js]: Yes, this is redundant!

File: examples/mocking/mocking.js (hunk 2 above)
Comment (joeyciechanowicz) [reply examples/mocking/mocking.js]: I was thinking that as it's an example we should show it being used all the time. But I think you're correct, that we should highlight that it's to be used in development only

File: src/mock-backend-response.js (hunk 3 above)
Comment (joeyciechanowicz) [reply src/mock-backend-response.js]: Yup, linter should have caught this. I'm used to `lint` running when I do `npm test`, but it was only for `npm run build`.

I've changed `lint` to run on `pretest` now to make sure it's caught in local devleopment.

**Pull request merged**